From 9925d09ad1d46505c151e08007e3846d94360714 Mon Sep 17 00:00:00 2001
From: Ricardo Vieira <ricardo.vieira@tecnico.ulisboa.pt>
Date: Tue, 27 May 2014 17:37:37 +0100
Subject: [PATCH] add vmarshal

---
 src/wayland-client.c | 12 ++++++++++++
 src/wayland-client.h |  3 +++
 2 files changed, 15 insertions(+)

diff --git a/src/wayland-client.c b/src/wayland-client.c
index e8aab7e..6bd56f2 100644
--- a/src/wayland-client.c
+++ b/src/wayland-client.c
@@ -550,6 +550,18 @@ wl_proxy_marshal_constructor(struct wl_proxy *proxy, uint32_t opcode,
 						  args, interface);
 }
 
+WL_EXPORT struct wl_proxy *
+wl_proxy_vmarshal_constructor(struct wl_proxy *proxy, uint32_t opcode,
+			     const struct wl_interface *interface, va_list ap)
+{
+	union wl_argument args[WL_CLOSURE_MAX_ARGS];
+	wl_argument_from_va_list(proxy->object.interface->methods[opcode].signature,
+				 args, WL_CLOSURE_MAX_ARGS, ap);
+
+	return wl_proxy_marshal_array_constructor(proxy, opcode,
+						  args, interface);
+}
+
 /** Prepare a request to be sent to the compositor
  *
  * \param proxy The proxy object
diff --git a/src/wayland-client.h b/src/wayland-client.h
index 2a32785..7d00846 100644
--- a/src/wayland-client.h
+++ b/src/wayland-client.h
@@ -134,6 +134,9 @@ struct wl_proxy *
 wl_proxy_marshal_array_constructor(struct wl_proxy *proxy,
 				   uint32_t opcode, union wl_argument *args,
 				   const struct wl_interface *interface);
+struct wl_proxy *
+wl_proxy_vmarshal_constructor(struct wl_proxy *proxy, uint32_t opcode,
+			     const struct wl_interface *interface, va_list ap);
 
 void wl_proxy_destroy(struct wl_proxy *proxy);
 int wl_proxy_add_listener(struct wl_proxy *proxy,
-- 
1.9.3

